diff -u -r cups-1.6.3.orig/config-scripts/cups-directories.m4 cups-1.6.3/config-scripts/cups-directories.m4
--- cups-1.6.3.orig/config-scripts/cups-directories.m4	2013-08-27 22:20:01.201998766 +0200
+++ cups-1.6.3/config-scripts/cups-directories.m4	2013-08-27 22:25:47.863998755 +0200
@@ -107,7 +107,7 @@
 			;;
 		Linux*)
 			if test -d /usr/lib64 -a ! -d /usr/lib64/fakeroot; then
-				libdir="$exec_prefix/lib64"
+				libdir="$exec_prefix/lib"
 			fi
 			;;
 		HP-UX*)
diff -u -r cups-1.6.3.orig/configure cups-1.6.3/configure
--- cups-1.6.3.orig/configure	2013-08-27 22:20:01.201998766 +0200
+++ cups-1.6.3/configure	2013-08-27 22:35:55.415998739 +0200
@@ -5844,7 +5844,7 @@
 			;;
 		Linux*)
 			if test -d /usr/lib64 -a ! -d /usr/lib64/fakeroot; then
-				libdir="$exec_prefix/lib64"
+				libdir="$exec_prefix/lib"
 			fi
 			;;
 		HP-UX*)
diff -u -r cups-1.6.3.orig/cups-config.in cups-1.6.3/cups-config.in
--- cups-1.6.3.orig/cups-config.in	2013-08-27 22:20:01.198998766 +0200
+++ cups-1.6.3/cups-config.in	2013-08-27 22:20:48.137998757 +0200
@@ -50,7 +50,7 @@
 	CFLAGS="$CFLAGS -I$includedir"
     fi
 
-    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64; then
+    if test $libdir != /usr/lib -a $libdir != /usr/lib -a $libdir != /usr/lib; then
 	LDFLAGS="$LDFLAGS -L$libdir"
     fi
 fi
